src/k8s/pkg/k8sd/controllers/node_configuration_test.go

```diff
@@ -109,21 +130,31 @@ func TestConfigPropagation(t *testing.T) {
 	})
 
 	go configController.Run(ctx)
-
 	defer watcher.Stop()
 
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
+			s.RestartServiceCalledWith = nil
+
 			g := NewWithT(t)
 
 			watcher.Add(tc.configmap)
+
+			// TODO: this is to ensure that the controller has handled the event. This should ideally
+			// be replaced with something like a "<-sentCh" instead
 			time.Sleep(100 * time.Millisecond)
 
-			for ekey, evalue := range tc.expectedUpdates {
+			for ekey, evalue := range tc.expectArgs {
 				val, err := snaputil.GetServiceArgument(s, "kubelet", ekey)
-				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(err).To(Succeed())
```

Comment:
I think succeed is only used for single return values. 
`Expect(err).To(Succeed())` reads wrong (although it might work)

https://github.com/onsi/gomega/blob/437995126acee58e6237f963913f54cb6386dd0a/matchers.go#L103